generate-ical.js

```diff
@@ -2,44 +2,156 @@
-        
+
+         // if tid = all then save file as calendar.ics else save file as tid.ics
+         if (eventType.tid === 'all') {
+            eventType.tid = 'calendar';
+        }
+
+        fs.writeFileSync(`./dist/${eventType.tid}.ics`, calendar.toString());
+
+    } catch (error) {
+        console.error(`Error generating calendar for type ${eventType.name}:`, error);
+    }
+}
+
+async function generateAllCalendars() {
+    try {
+        const eventTypes = await fetchEventTypes();
+
+        for (const eventType of eventTypes) {
+            await generateCalendarForType(eventType);
+        }
+
+        injectCalendarButtons(eventTypes);
+
     } catch (error) {
-        console.error('Error generating calendar:', error);
+        console.error('Error generating calendars:', error);
         process.exit(1);
     }
 }
 
-generateCalendar();
\ No newline at end of file
+function injectCalendarButtons(eventTypes) {
+    const distDir = './dist';
+    const files = fs.readdirSync(distDir);
+    const eventTypeMap = new Map(eventTypes.map(eventType => [eventType.tid, eventType.name]));
+
+    // Separate the "All/calendar" calendar file
+    const allFile = files.find(file => path.basename(file, '.ics') === 'calendar');
+    const otherFiles = files.filter(file => path.basename(file, '.ics') !== 'calendar').sort();
+
+    // Combine "All/calendar" file with sorted other files
+    const sortedFiles = allFile ? [allFile, ...otherFiles] : otherFiles;
+
+    const buttonsHtml = sortedFiles
+    .filter(file => file.endsWith('.ics'))
+    .map(file => {
+        const fileId = path.basename(file, '.ics');
+        const eventName = eventTypeMap.get(fileId) || fileId;
+        return `
+            <li>
+                <a href="webcal://miklb.github.io/tampa-events-ical/${file}" class="button">Subscribe to ${eventName} Calendar</a>
+                <a href="https://miklb.github.io/tampa-events-ical/${file}" download class="button" aria-label="Download ${eventName} Calendar" title="Download ${eventName} Calendar" role="button">
+                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" style="width: 1em; height: 1em; vertical-align: middle; fill: currentColor;">
+                        <path d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 242.7-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7 288 32zM64 352c-35.3 0-64 28.7-64 64l0 32c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-32c0-35.3-28.7-64-64-64l-101.5 0-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352 64 352zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"/>
+                    </svg>
+                </a>
+            </li>`;
+    })
+    .join('\n');
+
+    const indexPath = './index.html';
+    let indexHtml = fs.readFileSync(indexPath, 'utf8');
+    indexHtml = indexHtml.replace('<!-- CALENDAR_BUTTONS_PLACEHOLDER -->', `<ul>${buttonsHtml}</ul>`);
```

Comment:
Here is where that replacement happens.

Reply:
OK, thanks. I'll debug. Current build didn't execute that 

https://miklb.github.io/tampa-events-ical/.

 If you view source you see the  inline comment not buttons.
`<!-- CALENDAR_BUTTONS_PLACEHOLDER -->`

Reply:
Oh no sorry about that, maybe I copy pasted something wrong, thought I had tested it :(

Reply:
it might be in the deploy and the use of `./dist`

I'll double check the GH Action and the deploy. Thank you again for the improvements.

Reply:
I vaguely remember having to deploy from root for some reason and didn't want to change the path back. I'll debug.

Reply:
circling back there were two issues. There's a default action that deploys `/dist` without running the script (I probably need to remove that). That's what I was seeing after the merge. After running the the actual Action, and using the URL `https://miklb.github.io/tampa-events-ical/dist/` the new buttons are inserted. I'll update everything accordingly. Thank you again, this is a much better way to manage the subscriptions.